server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java

```diff
@@ -160,14 +159,11 @@ public static String createId(
         ByteUtils.writeLongLE(hash.h1, suffix, 0);
         ByteUtils.writeLongBE(timestamp, suffix, 8);   // Big Ending shrinks the inverted index by ~37%
 
-        String id = routingBuilder.createId(suffix, () -> {
-            if (dynamicMappersExists == false) {
-                throw new IllegalStateException(
-                    "Didn't find any fields to include in the routing which would be fine if there are"
-                        + " dynamic mapping waiting but we couldn't find any of those either!"
-                );
-            }
-            return 0;
+        String id = routingBuilder.createId(suffix, dynamicMappersExists ? () -> 0 : () -> {
+            throw new IllegalStateException(
```

Comment:
I think these lambdas can be static.

Reply:
They should automatically compile to code that reuses the same instance since they're non-capturing? No need to put them in a field I think?

Reply:
ah, ok!